lib/factory_bot/registry.rb

```diff
@@ -1,4 +1,20 @@
 module FactoryBot
+  # In Ruby <= 2.5 it is necessary to monkey patch
+  # the key error class because it does not expose
+  # receiver and key attributes.
+  # See https://bugs.ruby-lang.org/issues/14313
+  module KeyErrorExtension
+    attr_accessor :receiver, :key
+
+    def initialize(msg, key: nil, receiver: nil)
+      @key = key
+      @receiver = receiver
+      super msg
+    end
+  end
+
+  KeyError.prepend KeyErrorExtension
```

Comment:
Is there a way to do this only if it isn't already available? The built-in version will probably be a bit faster.
We could maybe check `KeyError.instance_method(:initialize).arity`, although that isn't the prettiest thing.

Reply:
What about:
```
Gem::Version.new(RUBY_VERSION) < Gem::Version.new('2.6')
```
?